Use `shutdown()` to exit on success.

davidlehn opened the pull request:
Depends on https://github.com/digitalbazaar/bedrock/pull/111.

davidlehn committed:
Use `shutdown()` for an orderly exit on success.

- Use `bedrock.shutdown()` for an orderly exit on success.
- The previous behavior was to abruptly shut down an app, even after
  testing success. Now event handlers will be called while app is
  shutting down:
  - `bedrock.stop`
  - `bedrock.stopped`
  - `bedrock.exit`

dlongley commented on CHANGELOG.md:
```diff
@@ -1,5 +1,23 @@
 # bedrock-test ChangeLog
 
+
+## 9.0.0 - 2024-xx-xx
```
```suggestion
## 9.0.0 - 2024-xx-xx
```

davidlehn commented:
This was written quite a while ago.  It was working for the use case I had.  It really could use more testing before a release. 
 I suspect it will work fine in most cases but would be good to check.  It's possible the handlers that were never called before, which now will be, could cause edge case issues.

davidlehn committed:
Upgrade dependencies.

- Upgrade dependencies:
  - `chai@6`
  - `chai-as-promised@8`
- Upgrade dev dependencies:
  - `c8@11`
